src/Columns/IColumnImpl.h

```diff
@@ -139,4 +139,41 @@ bool IColumn::hasEqualValuesImpl() const
     return true;
 }
 
+
+template <typename Derived>
+double IColumn::getRatioOfDefaultRowsImpl(double sample_ratio) const
+{
+    assert(sample_ratio > 0 && sample_ratio <= 1.0);
```

Comment:
it's better to throw logical error - not internal invariant, but developer's error